File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/detail/folder/FolderDetailUIState.kt
package dev.subfly.yabacore.state.detail.folder

import androidx.compose.runtime.Immutable
import dev.subfly.yabacore.model.ui.BookmarkUiModel
import dev.subfly.yabacore.model.ui.FolderUiModel
import dev.subfly.yabacore.model.utils.BookmarkAppearance
import dev.subfly.yabacore.model.utils.CardImageSizing
import dev.subfly.yabacore.model.utils.SortOrderType
import dev.subfly.yabacore.model.utils.SortType

@Immutable
data class FolderDetailUIState(
    val folder: FolderUiModel? = null,
    val query: String = "",
    val bookmarks: List<BookmarkUiModel> = emptyList(),
    val isSelectionMode: Boolean = false,
    val selectedBookmarkIds: Set<String> = emptySet(),
    val bookmarkAppearance: BookmarkAppearance = BookmarkAppearance.LIST,
    val cardImageSizing: CardImageSizing = CardImageSizing.SMALL,
    val sortType: SortType = SortType.CREATED_AT,
    val sortOrder: SortOrderType = SortOrderType.DESCENDING,
    val isLoading: Boolean = false,
)

File: Darwin/YABACore/State/Detail/Folder/FolderDetailUIState.swift
//
//  FolderDetailUIState.swift
//  YABACore
//

import Foundation

public struct FolderDetailUIState: Sendable {
    public var folderId: String?
    public var query: String
    public var selectionMode: Bool
    public var selectedBookmarkIds: Set<String>
    public var sortType: SortType
    public var sortOrder: SortOrderType
    public var bookmarkAppearance: BookmarkAppearance
    public var cardImageSizing: CardImageSizing?

    public init(
        folderId: String? = nil,
        query: String = "",
        selectionMode: Bool = false,
        selectedBookmarkIds: Set<String> = [],
        sortType: SortType = .editedAt,
        sortOrder: SortOrderType = .descending,
        bookmarkAppearance: BookmarkAppearance = .list,
        cardImageSizing: CardImageSizing? = .small
    ) {
        self.folderId = folderId
        self.query = query
        self.selectionMode = selectionMode
        self.selectedBookmarkIds = selectedBookmarkIds
        self.sortType = sortType
        self.sortOrder = sortOrder
        self.bookmarkAppearance = bookmarkAppearance
        self.cardImageSizing = cardImageSizing
    }
}
